### PR TITLE
Update cibuildwheel, add dependabot.

### PR DESCRIPTION
**Pull request by vtraag**
This is an attempt to update the `cibuildwheel` to the "official" PyPA version. It also adds a ["dependabot"](https://docs.github.com/en/code-security/dependabot/working-with-dependabot/automating-dependabot-with-github-actions), which automatically updates all dependencies in the GH Actions.

### CODE REVIEW

File: src/_igraph/graphobject.c [hunk 1]
```diff
@@ -13411,7 +13411,7 @@ struct PyMethodDef igraphmodule_Graph_methods[] = {
    "Generates a tree in which almost all vertices have the same number of children.\n\n"
    "@param n: the number of vertices in the graph\n"
    "@param children: the number of children of a vertex in the graph\n"
-   "@param type: determines whether the tree should be directed, and if\n"
+   "@param mode: determines whether the tree should be directed, and if\n"
```
Comment (ntamas) [inline src/_igraph/graphobject.c:13414]: Does this change also belong to this PR?

**Comment by vtraag** [comment]
Ah, no, sorry that was part of my previous PR that you already merged.

File: src/_igraph/graphobject.c (hunk 1 above)
Comment (vtraag) [reply src/_igraph/graphobject.c:13414]: Ah, no, sorry that was part of my previous PR that you already merged.

**Comment by ntamas** [comment]
Looks like we have a genuine problem with reference counting on Python 3.11, which will be hard to debug but we need to bite the bullet at some point. Failures related to refcounting on Python 3.11 should not prevent this PR from being merged.

**Comment by vtraag** [comment]
The ASan build doesn't reveal much more information. What would you like to do? Would you like to merge this while disabling the builds for 3.11 for the moment and opening a separate issue for the refcount problem? Or would your prefer to leave this PR open until we have solved the refcount problem?

**Pull request merged**